### ISSUE TITLE
apkmirror: missing icon extract

### ISSUE THREAD

**Issue by foxjaw** [ImranR98/Obtainium#1074]
"Track only" sourcing of host https://www.apkmirror.com has missing icon in obtainium menu.  

Icon URL format:  
`https://www.apkmirror.com/wp-content/themes/APKMirror/ap_resize/ap_resize.php?src=https%3A%2F%2Fdownloadr2.apkmirror.com%2Fwp-content%2Fuploads%2F2023%2F04%2F<app-id>.png&w=96&h=96&q=100`  

Example:  
https://www.apkmirror.com/wp-content/themes/APKMirror/ap_resize/ap_resize.php?src=https%3A%2F%2Fdownloadr2.apkmirror.com%2Fwp-content%2Fuploads%2F2023%2F04%2FMiXplorer_round.png&w=96&h=96&q=100

**Comment by ImranR98** [comment]
Obtainium doesn't extract icons from the source, it just grabs the icon from the  OS, so it only works for installed apps.  I'm pretty sure there is already an existing issue for extracting icons from the source.

**Comment by foxjaw** [comment]
No. The icon is missing even after app installation. It's only specific with apkmirror. If I remove apkmirror source & add some other like apkpure/aptoide/uptodown, all of them show the icon.  
Reconsider this once.

**Comment by ImranR98** [comment]
APKMirror is a track only source, so it uses pseudo IDs instead of real package IDs. So we cannot ask the OS for the app icon because it doesn't know what the real app ID is.  This could be solved if you can find a way to extract the real app ID from APK Mirror's website.

**Comment by foxjaw** [comment]
Oh k. How to extract the app id ? I'm taking the same example of [mixplorer](https://www.apkmirror.com/apk/hootan-parsa/mixplorer-hootanparsa/).

**Comment by ImranR98** [comment]
> How to extract the app id ?
If I knew I would have done it :)

 We try to extract the app ID from the website whenever possible, but when it's not possible we extract it from the APK itself. For APK mirror, neither option seems possible because we don't download the APK, and the website doesn't seem to have that information either.

**Comment by foxjaw** [comment]
Ah. Understood the issue. Is there a way to manually enter the app id ? I think that would fix this.

**Comment by foxjaw** [comment]
Btw, the metadata exists after [I select a variant](https://www.apkmirror.com/apk/hootan-parsa/mixplorer-hootanparsa/mixplorer-hootanparsa-6-64-3-release/mixplorer-6-64-3-android-apk-download/).

**Comment by ImranR98** [comment]
> Btw, the metadata exists after [I select a variant](https://www.apkmirror.com/apk/hootan-parsa/mixplorer-hootanparsa/mixplorer-hootanparsa-6-64-3-release/mixplorer-6-64-3-android-apk-download/).

Good catch! Missed that somehow.
